projects/modular_llm/get_clusters.py

```diff
@@ -0,0 +1,168 @@
+parser.add_argument(
+    "--new_dataname", type=str, default="zhan1993/flan-10k-flat-cluster-embedding"
+)
+
+parser.add_argument(
+    "--model", type=str, default="sentence-transformers/sentence-t5-xxl"
+)
+args = parser.parse_args()
+np.random.seed(42)
+
+
+def get_text_encode(text, model):
+
+    if args.encoding == "classifier":
+        return model.get_text_encode(text).cpu().detach().numpy()
+    elif args.encoding == "embedding":
+        return model.encode(text)
+
+
+if args.encoding == "classifier":
+    model = AdapterRankerHelper.get_ranker_instance(
+        ranker_model="classifier",
+        ranker_path="zhan1993/classifier_ranker_t5_v3",
+    )
+elif args.encoding == "embedding":
+    model = SentenceTransformer(args.model)
+
+# load the dataset
+
+
+def get_orca_dataset():
+
+    dataset = load_dataset("Open-Orca/OpenOrca")
+
+    # create the subsample of the dataset.
+    dataset_size = len(dataset["train"])
+    indices = list(range(dataset_size))
+    # random indices
+    np.random.shuffle(indices)
+    split = int(np.floor(args.subsample * dataset_size))
+    subset_indices = indices[:split]
+    subset_dataset = Subset(dataset["train"], subset_indices)
+
+    train_dataloader = DataLoader(
+        subset_dataset, batch_size=args.batch_size, num_workers=8
+    )
+    all_dataloader = DataLoader(
+        dataset["train"], batch_size=args.batch_size, num_workers=8
+    )
+
+    return train_dataloader, all_dataloader, dataset["train"]
+
+
+def get_flan_dataset():
+
+    flan = FlanModule(
+        FlanConfig(
+            model="EleutherAI/gpt-neo-125m",
+            model_family="gpt",
+            train_batch_size=4,
+            predict_batch_size=4,
+            dataset="sordonia/flan-10k-flat",
+            remove_phi_eval_tasks=True,
+        )
+    )
+
+    dataset = flan.train_dataset
+    # create the subsample of the dataset.
+    dataset_size = len(dataset)
+    indices = list(range(dataset_size))
+    # random indices
+    np.random.shuffle(indices)
+    split = int(np.floor(args.subsample * dataset_size))
+    subset_indices = indices[:split]
+    subset_dataset = Subset(dataset, subset_indices)
+
+    train_dataloader = DataLoader(
+        subset_dataset, batch_size=args.batch_size, num_workers=8
+    )
+    all_dataloader = flan.train_dataloader()
+
+    all_dataset = concatenate_datasets(
+        [flan.train_dataset, flan.dev_dataset, flan.test_dataset]
+    )
+
+    return train_dataloader, all_dataloader, all_dataset
+
+
+if __name__ == "__main__":
+    if args.dataset == "orca":
+        train_dataloader, all_dataloader, all_dataset = get_orca_dataset()
+    elif args.dataset == "flan":
+        train_dataloader, all_dataloader, all_dataset = get_flan_dataset()
+
+    embedding_list = []
+
+    for i, batch in tqdm(
+        enumerate(train_dataloader), total=len(train_dataloader), desc="dataset"
+    ):
+        if "source" in batch:
+            embedding = get_text_encode(batch["source"], model)
```

Comment:
can you add an argument from the command line that specifies which field to consider in the dataset?

--text_column_name "source"